src/lib.rs

```diff
@@ -177,22 +183,44 @@ impl State {
         // it creates new ones.
         // It's important to note that only creating new ones is async, otherwise it's just populating the friendships array
         while self.friendships.len() < amount_of_friendships {
-            let first_user;
-            let second_user;
+            let friendship = if used < available_friendships.len() {
+                let &(user1, user2) = &available_friendships[used];
+
+                let second_user_id_localpart =
+                    self.users.get(user2).unwrap().id().localpart().to_string();
+
+                let first_user = self.users.get_mut(user1).unwrap();
+                let first_user_id_localpart = first_user.id().localpart().to_string();
+                let homeserver = first_user.id().server_name().to_string();
+
+                let friendship = Friendship::from_ids(
+                    homeserver,
+                    first_user_id_localpart,
+                    second_user_id_localpart,
+                );
```

Comment:
I think we can derive friendship from user1 and user2 and avoid localpart recalculations.

Reply:
Now that user1 and user2 are the ids directly its even simpler